plugins/occt/module/vtkF3DOCCTReader.cxx

```diff
@@ -285,6 +315,104 @@ class vtkF3DOCCTReader::vtkInternals
   }
 
 #if F3D_PLUGIN_OCCT_XCAF
+  StyleMap CollectInheritedStyles(const TDF_Label& rootLabel, const TopoDS_Shape& rootShape)
+  {
+    StyleMap inheritedStyles;
+
+    if (rootLabel.IsNull())
+    {
+      return inheritedStyles;
+    }
+
+    /* collect styled shapes from the document */
+    StyleMap collectedStyles;
+    XCAFPrs::CollectStyleSettings(rootLabel, TopLoc_Location(), collectedStyles);
+
+    /* iterate styled shapes and assign style if leaf, otherwise collect if parent node */
+    const TopAbs_ShapeEnum leafType = this->Parent->GetReadWire() ? TopAbs_EDGE : TopAbs_FACE;
+    std::vector<std::pair<TopoDS_Shape, XCAFPrs_Style> > parents;
+    for (StyleMap::Iterator iter(collectedStyles); iter.More(); iter.Next())
+    {
+      const TopoDS_Shape& shape = iter.Key();
+      if (shape.ShapeType() == leafType)
+      {
+        inheritedStyles.Add(shape, iter.Value());
```

Comment:
do we need new data to cover this codepath ?

Reply:
I thought it would be reached, I'll have to double check why it's not and possibly remove it if it's not actually useful

Reply:
thanks for checking

Reply:
alright so it's a cute optimization in theory but it seems unlikely to come up IRL.

I think the optimization isn't worth the code complexity in the end (also it depends on the `ReadWire` state  [which is on by default](https://github.com/search?q=repo%3Af3d-app%2Ff3d%20ReadWireOn&type=code)), so I'll just skip it and keep it simple